### ISSUE TITLE
Update image scaling options

### ISSUE THREAD

**Issue by brivars** [cgeo/cgeo#14559]
### Tell us your idea!

Currently, when adding images to the logs, these scaling options are offered:
```
(No scaling)
512x384
640x480
800x600
1024x768
```
I think that this list is very outdated. Given how much the pixel density on mobile and other screens have grown, I think it would be safe to say no one really uses the smaller options but people probably would use bigger ones instead.

I propose removing the two smaller scaling options and adding two larger ones. The 800x600 can still be kept and IMO would be sufficiently compressed as a "data saving" option for slow connections/limited amount of data.

E.g. 1440x1080 & 2048x1536 or 2880x2160

### What solution would you suggest?

This could be the new scaling menu:
```
(No scaling)
800x600
1024x768
1440x1080
2048x1536
```

**Comment by eddiemuc** [comment]
What about "just" adding the two new larger options?

**Comment by fm-sys** [comment]
512x384 is really quite low. Have you never been in a situation where pictures have been posted where you can't see anything at all?

I always use 1024x768 which is totally fine for me, so 800x600 as minimum sounds reasonable.

Personally I don't need larger scalings, but I can very well understand the request. However, default should stay 1024x768 IMHO (AFAIK this is the current default, right?)

**Comment by eddiemuc** [comment]
> default should stay 1024x768 IMHO (AFAIK this is the current default, right?) 

IIRC the dialog remembers your last aelection and presents it as default next time you open it (on a "global" base, not per image or per cache).

**Comment by eddiemuc** [comment]
I checked code:
* The default scale option when the user never ever explicitely selected one is "no scaling"
* whenever the user explicitely selects a scale option within the "edit image" dialog (doesn't matter for which cache or image), then this new scaling will be the future defaultand preselection for any scaling
* to edit scaling options, it is enough to edit the log_image_scale_values array values in `strings_not_translateable.xml`. 
* The values set there will determine the LONGER side of the scaled image (for portrait it will be height, for landscape it will be width). The other side is calculated by c:geo such that image aspect ratio remains the same.
* Value "-1" means "no scaling"
* If a value is removed from the list in `strings_not_translateable.xml` but still used in some saved log images then this value will remain there. As soon as the "edit image" dialog is opened however, the "scale" will be set to first value (no scaling), and if this image is saved then this is the new scaling value

Current setting:
![image](https://github.com/cgeo/cgeo/assets/6909759/796f7bd7-5ff0-4373-a769-d8698c7b50a0)

**Comment by eddiemuc** [comment]
> 512x384 is really quite low. Have you never been in a situation where pictures have been posted where you can't see anything at all?

Yes, but imho this decision is up to the user. Anyway I have no strog opinion on it

**Comment by brivars** [comment]
> What about "just" adding the two new larger options?

Just adding is fine too. But I was thinking removing those to not make the selection too cluttered. And also really because those are too tiny to be practical.
Viewing on the website, 1024 is still an OK size, but as I can imagine it's probably on the phones where most people view images (looking for giveaway pics) and most phones these days have 2k or larger resolutions. So even 1024 appears tiny there. 
512x384 - you need a magnifying glass 😄 It's just not practical anymore

**Comment by ztNFny** [comment]
My vote goes to keeping the "small" option - I sometimes post photos with such a small size intentionally to avoid spoilers.

But how about replacing the dropdown with a slide that allows a custom size?

(Is there a max size for image upload on GC / OC?)

**Comment by SammysHP** [comment]
> But how about replacing the dropdown with a slide that allows a custom size?

Good idea. Could have the range from 512px to the long side of the image and if possible it could snap to common sizes.

But the distance might be too short to have good control over intermediate values. So what about common sizes from 512px to the long side of the selected image, but only with common sizes. If the user wants an intermediate size, a short tap on the value could open an input dialog (I've seen this many times in the past on Android).

**Comment by eddiemuc** [comment]
> (Is there a max size for image upload on GC / OC?)

Afaik there is, and our connector model even allows to specify one. But this value is currently not used. It could anyway only be used approximately because the actual image to upload (with the selected scaling) is only generated as part of the upload process. And there's no straight -forward way to calculate the size of a jpeg from given width and height (the values we display are only approximations as well, hence the usage of `~`)

**Comment by fm-sys** [comment]
> But how about replacing the dropdown with a slide that allows a custom size?

Please don't make a mountain out of a molehill. For sure is this possible. But do users have ever asked for it??? Another time I have the feeling we want to implement a completely useless feature which makes things more complicated for non power-users.

**Comment by ztNFny** [comment]
> > But how about replacing the dropdown with a slide that allows a custom size?
> 
> Please don't make a mountain out of a molehill. For sure is this possible. But do users have ever asked for it??? Another time I have the feeling we want to implement a completely useless feature which makes things more complicated for non power-users.

I *personally* find a slider a more intuitive GUI for resizing than a dropdown.

**Comment by eddiemuc** [comment]
I noticed a related strange behavior (bug?): when a log image is selected with a max(width, height)-value < 1024, then related scaling options will show the "unscaled" sizes. This is correct in the sense that our scaling will never upscale (means: will never make an image larger by scaling) but only downscale. Still it looks kinda strange.

For example see here the scaling list for am image with size 797x772px. The options "513" and "640" still scale, but the options "800" and "1024" don't. But this is not easy to grasp from the list...

![Screenshot_2023-08-29-18-41-40-63_a916ceb0c8a82619faeb76fe6b8ae997.jpg](https://github.com/cgeo/cgeo/assets/6909759/a6f34772-f451-486b-8b63-4bbaaba5f1b2)

This phenomenon will probably get more frequent when adding options 1440 and 2048

**Comment by fm-sys** [comment]
> > > But how about replacing the dropdown with a slide that allows a custom size?
> > 
> > 
> > Please don't make a mountain out of a molehill. For sure is this possible. But do users have ever asked for it??? Another time I have the feeling we want to implement a completely useless feature which makes things more complicated for non power-users.
> 
> I _personally_ find a slider a more intuitive GUI for resizing than a dropdown.

OK, let's concrete my opinion:

I slider can be a really nice GUI element if used in the right way. I have to admit that if we implement a slider with four or five equally distributed size options this would look just wonderful.

But please not a slider where I have to think about whether I choose 1400, 1440 or 1444px (yes, I'm the kind of guy who could spend hours thinking about the best number xD)

**Comment by brivars** [comment]
> My vote goes to keeping the "small" option - I sometimes post photos with such a small size intentionally to avoid spoilers.
> 
> But how about replacing the dropdown with a slide that allows a custom size?
> 
> (Is there a max size for image upload on GC / OC?)

If you post a photo that's so small others can't make things out in it intentionally, why post the photo at all? Would be my argument. If I post a photo I want others to actually be able to see it. If there are spoilers (that I don't want others to see) I crop them out or just not post it.

I would probably vote for a few preset options that are somewhat standard pixel resolutions, rather for a complete custom slider and images sizes being all over the place. But that is just me.
